packages/api-plugin-promotions-coupons/src/schemas/schema.graphql

```diff
@@ -42,11 +42,44 @@ type Coupon {
   discountId: ID
 }
 
+type CouponLog {
+  _id: ID!
+
+  "The shop ID"
+  shopId: ID!
+
+  "The coupon ID"
+  couponId: ID!
+
+  "The order ID"
+  orderId: ID
+
+  "The promotion ID"
+  promotionId: ID!
+
+  "The coupon owner ID"
+  accountId: ID
+
+  "The coupon code"
+  usedCount: Int
+
+  "The time the coupon was used"
+  createdAt: Date
+
+  "The log details for each time the coupon was used"
+  usedLogs: [JSONObject]
```

Comment:
Isn't this just going to be an array of a known entity? Seems like we could have a real shape of it defined?